RLBotPack/Noob_Black/src/bot.py

```diff
@@ -0,0 +1,528 @@
+from rlbot.agents.base_agent import BaseAgent, SimpleControllerState
+from rlbot.messages.flat.QuickChatSelection import QuickChatSelection
+from rlbot.utils.structures.game_data_struct import GameTickPacket
+
+from util.ball_prediction_analysis import find_slice_at_time
+from util.boost_pad_tracker import BoostPadTracker
+from util.drive import steer_toward_target
+from util.sequence import Sequence, ControlStep
+from util.vec import Vec3
+
+import math, time
+
+class MyBot(BaseAgent):
+
+    def __init__(self, name, team, index):
+        super().__init__(name, team, index)
+        self.active_sequence: Sequence = None
+        self.boost_pad_tracker = BoostPadTracker()
+        self.prev_jump = False
+        self.jump_tick = 0
+        self.prev_frame_nr = 0
+        self.prev_rot = [0, 0, 0]
+        self.prev_wait = False
+        self.current_attacker = -1
+
+    def initialize_agent(self):
+        self.boost_pad_tracker.initialize_boosts(self.get_field_info())
+
+    def get_output(self, packet: GameTickPacket) -> SimpleControllerState:
+        begin = time.time()
+        self.boost_pad_tracker.update_boost_status(packet)
+
+        if self.active_sequence is not None and not self.active_sequence.done:
+            controls = self.active_sequence.tick(packet)
+            if controls is not None:
+                return controls
+        # Get the amount of time taken before the next bounce
+        def next_bounce(pos, vel, height, g):
+            def solve_quadratic(a, b, c, ans, side):
```

Comment:
You could gain many performance improvements by moving these nested functions to the global scope instead.